{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ca81bc7f",
   "metadata": {},
   "source": [
    "# GTEx model with different priors"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be805cf1",
   "metadata": {},
   "source": [
    "## Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b4ed5e6f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "Loaded glmnet 4.1-10\n",
      "\n",
      "here() starts at /home/msubirana/Documents/pivlab/plier2-analyses\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!requireNamespace(\"PLIER\", quietly = TRUE)) {\n",
    "    devtools::install_github(\"wgmao/PLIER\")\n",
    "}\n",
    "\n",
    "# 3. Install PLIER2 (mchikina/PLIER2) if not already installed\n",
    "if (!requireNamespace(\"PLIER2\", quietly = TRUE)) {\n",
    "    REPO_PATH <- \"/home/msubirana/Documents/pivlab/PLIER2\"  # adjust\n",
    "    remotes::install_local(REPO_PATH, force = TRUE, dependencies = FALSE)\n",
    "}\n",
    "\n",
    "library(bigstatsr)\n",
    "library(data.table)\n",
    "library(dplyr)\n",
    "library(rsvd)\n",
    "library(glmnet)\n",
    "library(Matrix)\n",
    "library(knitr)\n",
    "library(here)\n",
    "library(PLIER2)\n",
    "\n",
    "source(here(\"config.R\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc858c79",
   "metadata": {},
   "source": [
    "## Output directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "02f368b4",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "output_data_dir <- config$GTEx$DATASET_FOLDER\n",
    "dir.create(output_data_dir, showWarnings = FALSE, recursive = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bab2d0f",
   "metadata": {},
   "source": [
    "# Load GTEx PLIER2 base model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bcb4c966",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gtex_baseRes <- readRDS(file.path(output_data_dir, \"gtex_PLIER2_baseRes.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "738b46b2",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gtex_svdRes <- readRDS(file.path(output_data_dir, \"gtex_svdRes.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4734948a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "PLIER_K_gtex <- readRDS(file.path(output_data_dir, \"PLIER_K_gtex.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5eb16174",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gtex_fbm_filt <- readRDS(file.path(output_data_dir, \"gtex_fbm_filt.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6beb6e7c",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gtex_genes <- readRDS(file.path(output_data_dir, \"gtex_genes.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9ae9d777",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "samples <- readRDS(file.path(output_data_dir, \"gtex_samples.rds\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ee267bb",
   "metadata": {},
   "source": [
    "## Prior KEGG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35ea5a8c",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      ">>> Running PLIER for prior: KEGG_2021_Human\n",
      "\n",
      "Auto-detected name: KEGG_2021_Human\n",
      "\n",
      "Using cached file for KEGG_2021_Human\n",
      "\n",
      "There are 6409 genes in the intersection between data and prior\n",
      "\n",
      "Removing 10 pathways\n",
      "\n",
      "Inverting...\n",
      "\n",
      "done\n",
      "\n",
      "**PLIER v2 **\n",
      "\n",
      "Warning message:\n",
      "“`seed` is deprecated and ignored. Use set.seed(seed) before calling this function.”\n",
      "using provided PLIERbase result\n",
      "\n",
      "L1=45.2778145717356; L2=135.833443715207\n",
      "\n",
      "Progress 1 / 350 | Bdiff=0.000413\n",
      "\n",
      "Progress 2 / 350 | Bdiff=0.000367\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 3 / 350 | Bdiff=0.000338\n",
      "\n",
      "Progress 4 / 350 | Bdiff=0.000299\n",
      "\n",
      ", Number of annotated columns is 302\n",
      "\n",
      "Progress 5 / 350 | Bdiff=0.000278\n",
      "\n",
      "Progress 6 / 350 | Bdiff=0.000250\n",
      "\n",
      ", Number of annotated columns is 302\n",
      "\n",
      "Progress 7 / 350 | Bdiff=0.000231\n",
      "\n",
      "Progress 8 / 350 | Bdiff=0.000209\n",
      "\n",
      ", Number of annotated columns is 304\n",
      "\n",
      "Progress 9 / 350 | Bdiff=0.000191\n",
      "\n",
      "Progress 10 / 350 | Bdiff=0.000188\n",
      "\n",
      ", Number of annotated columns is 306\n",
      "\n",
      "Progress 11 / 350 | Bdiff=0.000179\n",
      "\n",
      "Progress 12 / 350 | Bdiff=0.000174\n",
      "\n",
      ", Number of annotated columns is 300\n",
      "\n",
      "Progress 13 / 350 | Bdiff=0.000166\n",
      "\n",
      "Progress 14 / 350 | Bdiff=0.000167\n",
      "\n",
      ", Number of annotated columns is 301\n",
      "\n",
      "Progress 15 / 350 | Bdiff=0.000160\n",
      "\n",
      "Progress 16 / 350 | Bdiff=0.000158\n",
      "\n",
      ", Number of annotated columns is 302\n",
      "\n",
      "Progress 17 / 350 | Bdiff=0.000154\n",
      "\n",
      "Progress 18 / 350 | Bdiff=0.000152\n",
      "\n",
      ", Number of annotated columns is 302\n",
      "\n",
      "Progress 19 / 350 | Bdiff=0.000146\n",
      "\n",
      "Progress 20 / 350 | Bdiff=0.000139\n",
      "\n",
      ", Number of annotated columns is 302\n",
      "\n",
      "Progress 21 / 350 | Bdiff=0.000131\n",
      "\n",
      "Progress 22 / 350 | Bdiff=0.000128\n",
      "\n",
      ", Number of annotated columns is 294\n",
      "\n",
      "Progress 23 / 350 | Bdiff=0.000126\n",
      "\n",
      "Progress 24 / 350 | Bdiff=0.000120\n",
      "\n",
      ", Number of annotated columns is 295\n",
      "\n",
      "Progress 25 / 350 | Bdiff=0.000120\n",
      "\n",
      "Progress 26 / 350 | Bdiff=0.000108\n",
      "\n",
      ", Number of annotated columns is 296\n",
      "\n",
      "Progress 27 / 350 | Bdiff=0.000103\n",
      "\n",
      "Progress 28 / 350 | Bdiff=0.000100\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 29 / 350 | Bdiff=0.000095\n",
      "\n",
      "Progress 30 / 350 | Bdiff=0.000095\n",
      "\n",
      ", Number of annotated columns is 300\n",
      "\n",
      "Progress 31 / 350 | Bdiff=0.000096\n",
      "\n",
      "Progress 32 / 350 | Bdiff=0.000093\n",
      "\n",
      ", Number of annotated columns is 293\n",
      "\n",
      "Progress 33 / 350 | Bdiff=0.000086\n",
      "\n",
      "Progress 34 / 350 | Bdiff=0.000076\n",
      "\n",
      ", Number of annotated columns is 295\n",
      "\n",
      "Progress 35 / 350 | Bdiff=0.000076\n",
      "\n",
      "Progress 36 / 350 | Bdiff=0.000071\n",
      "\n",
      ", Number of annotated columns is 296\n",
      "\n",
      "Progress 37 / 350 | Bdiff=0.000071\n",
      "\n",
      "Progress 38 / 350 | Bdiff=0.000066\n",
      "\n",
      ", Number of annotated columns is 296\n",
      "\n",
      "Progress 39 / 350 | Bdiff=0.000060\n",
      "\n",
      "Progress 40 / 350 | Bdiff=0.000060\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 41 / 350 | Bdiff=0.000057\n",
      "\n",
      "Progress 42 / 350 | Bdiff=0.000055\n",
      "\n",
      ", Number of annotated columns is 293\n",
      "\n",
      "Progress 43 / 350 | Bdiff=0.000051\n",
      "\n",
      "Progress 44 / 350 | Bdiff=0.000050\n",
      "\n",
      ", Number of annotated columns is 293\n",
      "\n",
      "Progress 45 / 350 | Bdiff=0.000048\n",
      "\n",
      "Progress 46 / 350 | Bdiff=0.000046\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 47 / 350 | Bdiff=0.000044\n",
      "\n",
      "Progress 48 / 350 | Bdiff=0.000041\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 49 / 350 | Bdiff=0.000040\n",
      "\n",
      "Progress 50 / 350 | Bdiff=0.000040\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 51 / 350 | Bdiff=0.000041\n",
      "\n",
      "Progress 52 / 350 | Bdiff=0.000040\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 53 / 350 | Bdiff=0.000038\n",
      "\n",
      "Progress 54 / 350 | Bdiff=0.000036\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 55 / 350 | Bdiff=0.000035\n",
      "\n",
      "Progress 56 / 350 | Bdiff=0.000035\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 57 / 350 | Bdiff=0.000035\n",
      "\n",
      "Progress 58 / 350 | Bdiff=0.000032\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 59 / 350 | Bdiff=0.000031\n",
      "\n",
      "Progress 60 / 350 | Bdiff=0.000031\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 61 / 350 | Bdiff=0.000031\n",
      "\n",
      "Progress 62 / 350 | Bdiff=0.000031\n",
      "\n",
      ", Number of annotated columns is 295\n",
      "\n",
      "Progress 63 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 64 / 350 | Bdiff=0.000031\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 65 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 66 / 350 | Bdiff=0.000032\n",
      "\n",
      ", Number of annotated columns is 300\n",
      "\n",
      "Progress 67 / 350 | Bdiff=0.000034\n",
      "\n",
      "Progress 68 / 350 | Bdiff=0.000035\n",
      "\n",
      ", Number of annotated columns is 300\n",
      "\n",
      "Progress 69 / 350 | Bdiff=0.000034\n",
      "\n",
      "Progress 70 / 350 | Bdiff=0.000035\n",
      "\n",
      ", Number of annotated columns is 300\n",
      "\n",
      "Progress 71 / 350 | Bdiff=0.000037\n",
      "\n",
      "Progress 72 / 350 | Bdiff=0.000035\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 73 / 350 | Bdiff=0.000036\n",
      "\n",
      "Progress 74 / 350 | Bdiff=0.000034\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 75 / 350 | Bdiff=0.000037\n",
      "\n",
      "Progress 76 / 350 | Bdiff=0.000040\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 77 / 350 | Bdiff=0.000042\n",
      "\n",
      "Progress 78 / 350 | Bdiff=0.000041\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 79 / 350 | Bdiff=0.000042\n",
      "\n",
      "Progress 80 / 350 | Bdiff=0.000048\n",
      "\n",
      ", Number of annotated columns is 299\n",
      "\n",
      "Progress 81 / 350 | Bdiff=0.000046\n",
      "\n",
      "Progress 82 / 350 | Bdiff=0.000046\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 83 / 350 | Bdiff=0.000050\n",
      "\n",
      "Progress 84 / 350 | Bdiff=0.000047\n",
      "\n",
      ", Number of annotated columns is 297\n",
      "\n",
      "Progress 85 / 350 | Bdiff=0.000052\n",
      "\n",
      "Progress 86 / 350 | Bdiff=0.000054\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 87 / 350 | Bdiff=0.000059\n",
      "\n",
      "Progress 88 / 350 | Bdiff=0.000063\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 89 / 350 | Bdiff=0.000065\n",
      "\n",
      "Progress 90 / 350 | Bdiff=0.000065\n",
      "\n",
      ", Number of annotated columns is 298\n",
      "\n",
      "Progress 91 / 350 | Bdiff=0.000066\n",
      "\n",
      "Progress 92 / 350 | Bdiff=0.000069\n",
      "\n",
      ", Number of annotated columns is 295\n",
      "\n",
      "Progress 93 / 350 | Bdiff=0.000068\n",
      "\n",
      "Progress 94 / 350 | Bdiff=0.000070\n",
      "\n",
      "converged at iteration94Bdiff is not decreasing\n",
      "\n",
      "Updating Z for CV\n",
      "\n",
      "crossValidation\n",
      "\n",
      "There are 166 LVs with AUC>0.70\n",
      "\n",
      "There are 34 LVs with AUC>0.90\n",
      "\n",
      "Saved: /home/msubirana/Documents/pivlab/plier2-analyses/output/gtex/gtex_KEGG_2021_Human_PLIER2.rds\n",
      "\n",
      ">>> Running PLIER for prior: GWAS_Catalog_2025\n",
      "\n",
      "Auto-detected name: GWAS_Catalog_2025\n",
      "\n",
      "Downloading GWAS_Catalog_2025 from Enrichr...\n",
      "\n",
      "There are 12382 genes in the intersection between data and prior\n",
      "\n",
      "Removing 1106 pathways\n",
      "\n",
      "Inverting...\n",
      "\n",
      "done\n",
      "\n",
      "**PLIER v2 **\n",
      "\n",
      "Warning message:\n",
      "“`seed` is deprecated and ignored. Use set.seed(seed) before calling this function.”\n",
      "using provided PLIERbase result\n",
      "\n",
      "L1=45.2778145717356; L2=135.833443715207\n",
      "\n",
      "Progress 1 / 350 | Bdiff=0.000413\n",
      "\n",
      "Progress 2 / 350 | Bdiff=0.000367\n",
      "\n",
      ", Number of annotated columns is 215\n",
      "\n",
      "Progress 3 / 350 | Bdiff=0.000334\n",
      "\n",
      "Progress 4 / 350 | Bdiff=0.000300\n",
      "\n",
      ", Number of annotated columns is 219\n",
      "\n",
      "Progress 5 / 350 | Bdiff=0.000278\n",
      "\n",
      "Progress 6 / 350 | Bdiff=0.000248\n",
      "\n",
      ", Number of annotated columns is 223\n",
      "\n",
      "Progress 7 / 350 | Bdiff=0.000232\n",
      "\n",
      "Progress 8 / 350 | Bdiff=0.000209\n",
      "\n",
      ", Number of annotated columns is 224\n",
      "\n",
      "Progress 9 / 350 | Bdiff=0.000190\n",
      "\n",
      "Progress 10 / 350 | Bdiff=0.000188\n",
      "\n",
      ", Number of annotated columns is 225\n",
      "\n",
      "Progress 11 / 350 | Bdiff=0.000179\n",
      "\n",
      "Progress 12 / 350 | Bdiff=0.000173\n",
      "\n",
      ", Number of annotated columns is 221\n",
      "\n",
      "Progress 13 / 350 | Bdiff=0.000163\n",
      "\n",
      "Progress 14 / 350 | Bdiff=0.000167\n",
      "\n",
      ", Number of annotated columns is 223\n",
      "\n",
      "Progress 15 / 350 | Bdiff=0.000163\n",
      "\n",
      "Progress 16 / 350 | Bdiff=0.000155\n",
      "\n",
      ", Number of annotated columns is 225\n",
      "\n",
      "Progress 17 / 350 | Bdiff=0.000154\n",
      "\n",
      "Progress 18 / 350 | Bdiff=0.000147\n",
      "\n",
      ", Number of annotated columns is 227\n",
      "\n",
      "Progress 19 / 350 | Bdiff=0.000146\n",
      "\n",
      "Progress 20 / 350 | Bdiff=0.000143\n",
      "\n",
      ", Number of annotated columns is 229\n",
      "\n",
      "Progress 21 / 350 | Bdiff=0.000131\n",
      "\n",
      "Progress 22 / 350 | Bdiff=0.000130\n",
      "\n",
      ", Number of annotated columns is 224\n",
      "\n",
      "Progress 23 / 350 | Bdiff=0.000127\n",
      "\n",
      "Progress 24 / 350 | Bdiff=0.000119\n",
      "\n",
      ", Number of annotated columns is 226\n",
      "\n",
      "Progress 25 / 350 | Bdiff=0.000121\n",
      "\n",
      "Progress 26 / 350 | Bdiff=0.000114\n",
      "\n",
      ", Number of annotated columns is 231\n",
      "\n",
      "Progress 27 / 350 | Bdiff=0.000102\n",
      "\n",
      "Progress 28 / 350 | Bdiff=0.000101\n",
      "\n",
      ", Number of annotated columns is 233\n",
      "\n",
      "Progress 29 / 350 | Bdiff=0.000100\n",
      "\n",
      "Progress 30 / 350 | Bdiff=0.000098\n",
      "\n",
      ", Number of annotated columns is 233\n",
      "\n",
      "Progress 31 / 350 | Bdiff=0.000094\n",
      "\n",
      "Progress 32 / 350 | Bdiff=0.000089\n",
      "\n",
      ", Number of annotated columns is 226\n",
      "\n",
      "Progress 33 / 350 | Bdiff=0.000082\n",
      "\n",
      "Progress 34 / 350 | Bdiff=0.000076\n",
      "\n",
      ", Number of annotated columns is 226\n",
      "\n",
      "Progress 35 / 350 | Bdiff=0.000074\n",
      "\n",
      "Progress 36 / 350 | Bdiff=0.000075\n",
      "\n",
      ", Number of annotated columns is 228\n",
      "\n",
      "Progress 37 / 350 | Bdiff=0.000071\n",
      "\n",
      "Progress 38 / 350 | Bdiff=0.000071\n",
      "\n",
      ", Number of annotated columns is 229\n",
      "\n",
      "Progress 39 / 350 | Bdiff=0.000061\n",
      "\n",
      "Progress 40 / 350 | Bdiff=0.000059\n",
      "\n",
      ", Number of annotated columns is 232\n",
      "\n",
      "Progress 41 / 350 | Bdiff=0.000058\n",
      "\n",
      "Progress 42 / 350 | Bdiff=0.000055\n",
      "\n",
      ", Number of annotated columns is 224\n",
      "\n",
      "Progress 43 / 350 | Bdiff=0.000054\n",
      "\n",
      "Progress 44 / 350 | Bdiff=0.000048\n",
      "\n",
      ", Number of annotated columns is 225\n",
      "\n",
      "Progress 45 / 350 | Bdiff=0.000046\n",
      "\n",
      "Progress 46 / 350 | Bdiff=0.000046\n",
      "\n",
      ", Number of annotated columns is 225\n",
      "\n",
      "Progress 47 / 350 | Bdiff=0.000046\n",
      "\n",
      "Progress 48 / 350 | Bdiff=0.000043\n",
      "\n",
      ", Number of annotated columns is 225\n",
      "\n",
      "Progress 49 / 350 | Bdiff=0.000043\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Vector of prior URLs\n",
    "prior_urls <- c(\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=KEGG_2021_Human\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GWAS_Catalog_2025\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GO_Biological_Process_2025\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GTEx_Tissues_V8_2023\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Chromosome_Location\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Human_Gene_Atlas\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=LINCS_L1000_CRISPR_KO_Consensus_Sigs\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Metabolomics_Workbench_Metabolites_2022\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=OMIM_Disease\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Proteomics_Drug_Atlas_2023\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=TF_Perturbations_Followed_by_Expression\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=UK_Biobank_GWAS_v1\",\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=CellMarker_2024\"\n",
    ")\n",
    "\n",
    "# Names are exactly what's after libraryName=\n",
    "names(prior_urls) <- sub(\".*libraryName=\", \"\", prior_urls)\n",
    "\n",
    "results <- list()\n",
    "\n",
    "for (prior_name in names(prior_urls)) {\n",
    "  url <- prior_urls[[prior_name]]\n",
    "  message(\">>> Running PLIER for prior: \", prior_name)\n",
    "\n",
    "  try({\n",
    "    gmt <- getGMT(url)\n",
    "    gmt_list <- list()\n",
    "    gmt_list[[prior_name]] <- gmt\n",
    "\n",
    "    pathMat   <- gmtListToSparseMat(gmt_list)\n",
    "    matched   <- getMatchedPathwayMat(pathMat, gtex_genes)\n",
    "    if (is.null(matched) || ncol(matched) == 0) {\n",
    "      warning(\"No matched pathways for \", prior_name, \"; skipping.\")\n",
    "      next\n",
    "    }\n",
    "    chatObj   <- getChat(matched)\n",
    "\n",
    "    fullRes <- PLIERfull(\n",
    "      Y                 = gtex_fbm_filt,\n",
    "      priorMat          = as.matrix(matched),\n",
    "      svdres            = gtex_svdRes,\n",
    "      plier.base.result = gtex_baseRes,\n",
    "      Chat              = chatObj,\n",
    "      k                 = PLIER_K_gtex,\n",
    "      doCrossval        = TRUE,\n",
    "      trace             = TRUE,\n",
    "      max.U.updates     = config$GTEx$MAX_U_UPDATES\n",
    "    )\n",
    "\n",
    "    if (!is.null(fullRes$B)) colnames(fullRes$B) <- samples\n",
    "    if (!is.null(fullRes$Z)) colnames(fullRes$Z) <- paste0(\"LV\", seq_len(ncol(fullRes$Z)))\n",
    "    if (!is.null(fullRes$summary)) {\n",
    "      fullRes$summary <- fullRes$summary |>\n",
    "        dplyr::rename(LV = LV_index) |>\n",
    "        dplyr::mutate(LV = paste0(\"LV\", LV))\n",
    "    }\n",
    "\n",
    "    out_file <- file.path(output_data_dir, sprintf(\"gtex_%s_PLIER2.rds\", prior_name))\n",
    "    saveRDS(fullRes, file = out_file)\n",
    "    results[[prior_name]] <- fullRes\n",
    "\n",
    "    message(\"Saved: \", out_file)\n",
    "  }, silent = FALSE)\n",
    "}\n",
    "\n",
    "invisible(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "af6f714f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      ">>> Running PLIER for prior: CellMarker_2024\n",
      "\n",
      "Auto-detected name: CellMarker_2024\n",
      "\n",
      "Using cached file for CellMarker_2024\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There are 11124 genes in the intersection between data and prior\n",
      "\n",
      "Removing 965 pathways\n",
      "\n",
      "Inverting...\n",
      "\n",
      "done\n",
      "\n",
      "**PLIER v2 **\n",
      "\n",
      "Warning message:\n",
      "“`seed` is deprecated and ignored. Use set.seed(seed) before calling this function.”\n",
      "using provided PLIERbase result\n",
      "\n",
      "L1=45.2778145717356; L2=135.833443715207\n",
      "\n",
      "Progress 1 / 350 | Bdiff=0.000413\n",
      "\n",
      "Progress 2 / 350 | Bdiff=0.000367\n",
      "\n",
      ", Number of annotated columns is 267\n",
      "\n",
      "Progress 3 / 350 | Bdiff=0.000340\n",
      "\n",
      "Progress 4 / 350 | Bdiff=0.000302\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 5 / 350 | Bdiff=0.000279\n",
      "\n",
      "Progress 6 / 350 | Bdiff=0.000247\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 7 / 350 | Bdiff=0.000234\n",
      "\n",
      "Progress 8 / 350 | Bdiff=0.000212\n",
      "\n",
      ", Number of annotated columns is 272\n",
      "\n",
      "Progress 9 / 350 | Bdiff=0.000196\n",
      "\n",
      "Progress 10 / 350 | Bdiff=0.000188\n",
      "\n",
      ", Number of annotated columns is 275\n",
      "\n",
      "Progress 11 / 350 | Bdiff=0.000179\n",
      "\n",
      "Progress 12 / 350 | Bdiff=0.000175\n",
      "\n",
      ", Number of annotated columns is 272\n",
      "\n",
      "Progress 13 / 350 | Bdiff=0.000162\n",
      "\n",
      "Progress 14 / 350 | Bdiff=0.000165\n",
      "\n",
      ", Number of annotated columns is 273\n",
      "\n",
      "Progress 15 / 350 | Bdiff=0.000159\n",
      "\n",
      "Progress 16 / 350 | Bdiff=0.000156\n",
      "\n",
      ", Number of annotated columns is 276\n",
      "\n",
      "Progress 17 / 350 | Bdiff=0.000154\n",
      "\n",
      "Progress 18 / 350 | Bdiff=0.000151\n",
      "\n",
      ", Number of annotated columns is 277\n",
      "\n",
      "Progress 19 / 350 | Bdiff=0.000146\n",
      "\n",
      "Progress 20 / 350 | Bdiff=0.000141\n",
      "\n",
      ", Number of annotated columns is 280\n",
      "\n",
      "Progress 21 / 350 | Bdiff=0.000132\n",
      "\n",
      "Progress 22 / 350 | Bdiff=0.000128\n",
      "\n",
      ", Number of annotated columns is 272\n",
      "\n",
      "Progress 23 / 350 | Bdiff=0.000129\n",
      "\n",
      "Progress 24 / 350 | Bdiff=0.000122\n",
      "\n",
      ", Number of annotated columns is 276\n",
      "\n",
      "Progress 25 / 350 | Bdiff=0.000118\n",
      "\n",
      "Progress 26 / 350 | Bdiff=0.000107\n",
      "\n",
      ", Number of annotated columns is 277\n",
      "\n",
      "Progress 27 / 350 | Bdiff=0.000100\n",
      "\n",
      "Progress 28 / 350 | Bdiff=0.000101\n",
      "\n",
      ", Number of annotated columns is 278\n",
      "\n",
      "Progress 29 / 350 | Bdiff=0.000097\n",
      "\n",
      "Progress 30 / 350 | Bdiff=0.000095\n",
      "\n",
      ", Number of annotated columns is 278\n",
      "\n",
      "Progress 31 / 350 | Bdiff=0.000093\n",
      "\n",
      "Progress 32 / 350 | Bdiff=0.000089\n",
      "\n",
      ", Number of annotated columns is 277\n",
      "\n",
      "Progress 33 / 350 | Bdiff=0.000086\n",
      "\n",
      "Progress 34 / 350 | Bdiff=0.000078\n",
      "\n",
      ", Number of annotated columns is 280\n",
      "\n",
      "Progress 35 / 350 | Bdiff=0.000074\n",
      "\n",
      "Progress 36 / 350 | Bdiff=0.000071\n",
      "\n",
      ", Number of annotated columns is 280\n",
      "\n",
      "Progress 37 / 350 | Bdiff=0.000070\n",
      "\n",
      "Progress 38 / 350 | Bdiff=0.000064\n",
      "\n",
      ", Number of annotated columns is 281\n",
      "\n",
      "Progress 39 / 350 | Bdiff=0.000062\n",
      "\n",
      "Progress 40 / 350 | Bdiff=0.000058\n",
      "\n",
      ", Number of annotated columns is 284\n",
      "\n",
      "Progress 41 / 350 | Bdiff=0.000055\n",
      "\n",
      "Progress 42 / 350 | Bdiff=0.000052\n",
      "\n",
      ", Number of annotated columns is 275\n",
      "\n",
      "Progress 43 / 350 | Bdiff=0.000051\n",
      "\n",
      "Progress 44 / 350 | Bdiff=0.000047\n",
      "\n",
      ", Number of annotated columns is 276\n",
      "\n",
      "Progress 45 / 350 | Bdiff=0.000046\n",
      "\n",
      "Progress 46 / 350 | Bdiff=0.000044\n",
      "\n",
      ", Number of annotated columns is 277\n",
      "\n",
      "Progress 47 / 350 | Bdiff=0.000042\n",
      "\n",
      "Progress 48 / 350 | Bdiff=0.000038\n",
      "\n",
      ", Number of annotated columns is 277\n",
      "\n",
      "Progress 49 / 350 | Bdiff=0.000042\n",
      "\n",
      "Progress 50 / 350 | Bdiff=0.000042\n",
      "\n",
      ", Number of annotated columns is 278\n",
      "\n",
      "Progress 51 / 350 | Bdiff=0.000040\n",
      "\n",
      "Progress 52 / 350 | Bdiff=0.000040\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 53 / 350 | Bdiff=0.000041\n",
      "\n",
      "Progress 54 / 350 | Bdiff=0.000039\n",
      "\n",
      ", Number of annotated columns is 272\n",
      "\n",
      "Progress 55 / 350 | Bdiff=0.000038\n",
      "\n",
      "Progress 56 / 350 | Bdiff=0.000035\n",
      "\n",
      ", Number of annotated columns is 273\n",
      "\n",
      "Progress 57 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 58 / 350 | Bdiff=0.000033\n",
      "\n",
      ", Number of annotated columns is 274\n",
      "\n",
      "Progress 59 / 350 | Bdiff=0.000033\n",
      "\n",
      "Progress 60 / 350 | Bdiff=0.000032\n",
      "\n",
      ", Number of annotated columns is 276\n",
      "\n",
      "Progress 61 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 62 / 350 | Bdiff=0.000030\n",
      "\n",
      ", Number of annotated columns is 272\n",
      "\n",
      "Progress 63 / 350 | Bdiff=0.000034\n",
      "\n",
      "Progress 64 / 350 | Bdiff=0.000034\n",
      "\n",
      ", Number of annotated columns is 273\n",
      "\n",
      "Progress 65 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 66 / 350 | Bdiff=0.000031\n",
      "\n",
      ", Number of annotated columns is 273\n",
      "\n",
      "Progress 67 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 68 / 350 | Bdiff=0.000033\n",
      "\n",
      ", Number of annotated columns is 274\n",
      "\n",
      "Progress 69 / 350 | Bdiff=0.000032\n",
      "\n",
      "Progress 70 / 350 | Bdiff=0.000034\n",
      "\n",
      ", Number of annotated columns is 274\n",
      "\n",
      "Progress 71 / 350 | Bdiff=0.000033\n",
      "\n",
      "Progress 72 / 350 | Bdiff=0.000033\n",
      "\n",
      ", Number of annotated columns is 267\n",
      "\n",
      "Progress 73 / 350 | Bdiff=0.000036\n",
      "\n",
      "Progress 74 / 350 | Bdiff=0.000037\n",
      "\n",
      ", Number of annotated columns is 267\n",
      "\n",
      "Progress 75 / 350 | Bdiff=0.000037\n",
      "\n",
      "Progress 76 / 350 | Bdiff=0.000038\n",
      "\n",
      ", Number of annotated columns is 268\n",
      "\n",
      "Progress 77 / 350 | Bdiff=0.000037\n",
      "\n",
      "Progress 78 / 350 | Bdiff=0.000037\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 79 / 350 | Bdiff=0.000038\n",
      "\n",
      "Progress 80 / 350 | Bdiff=0.000036\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 81 / 350 | Bdiff=0.000038\n",
      "\n",
      "Progress 82 / 350 | Bdiff=0.000039\n",
      "\n",
      ", Number of annotated columns is 267\n",
      "\n",
      "Progress 83 / 350 | Bdiff=0.000039\n",
      "\n",
      "Progress 84 / 350 | Bdiff=0.000046\n",
      "\n",
      ", Number of annotated columns is 268\n",
      "\n",
      "Progress 85 / 350 | Bdiff=0.000049\n",
      "\n",
      "Progress 86 / 350 | Bdiff=0.000048\n",
      "\n",
      ", Number of annotated columns is 268\n",
      "\n",
      "Progress 87 / 350 | Bdiff=0.000050\n",
      "\n",
      "Progress 88 / 350 | Bdiff=0.000053\n",
      "\n",
      ", Number of annotated columns is 269\n",
      "\n",
      "Progress 89 / 350 | Bdiff=0.000054\n",
      "\n",
      "Progress 90 / 350 | Bdiff=0.000055\n",
      "\n",
      ", Number of annotated columns is 271\n",
      "\n",
      "Progress 91 / 350 | Bdiff=0.000059\n",
      "\n",
      "Progress 92 / 350 | Bdiff=0.000064\n",
      "\n",
      ", Number of annotated columns is 265\n",
      "\n",
      "Progress 93 / 350 | Bdiff=0.000067\n",
      "\n",
      "Progress 94 / 350 | Bdiff=0.000067\n",
      "\n",
      ", Number of annotated columns is 266\n",
      "\n",
      "Progress 95 / 350 | Bdiff=0.000066\n",
      "\n",
      "Progress 96 / 350 | Bdiff=0.000065\n",
      "\n",
      "converged at iteration96Bdiff is not decreasing\n",
      "\n",
      "Updating Z for CV\n",
      "\n",
      "crossValidation\n",
      "\n",
      "There are 203 LVs with AUC>0.70\n",
      "\n",
      "There are 107 LVs with AUC>0.90\n",
      "\n",
      "Saved: /home/msubirana/Documents/pivlab/plier2-analyses/output/gtex/gtex_CellMarker_2024_PLIER2.rds\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Vector of prior URLs\n",
    "prior_urls <- c(\n",
    "  \"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=CellMarker_2024\"\n",
    ")\n",
    "\n",
    "# Names are exactly what's after libraryName=\n",
    "names(prior_urls) <- sub(\".*libraryName=\", \"\", prior_urls)\n",
    "\n",
    "results <- list()\n",
    "\n",
    "for (prior_name in names(prior_urls)) {\n",
    "  url <- prior_urls[[prior_name]]\n",
    "  message(\">>> Running PLIER for prior: \", prior_name)\n",
    "\n",
    "  try({\n",
    "    gmt <- getGMT(url)\n",
    "    gmt_list <- list()\n",
    "    gmt_list[[prior_name]] <- gmt\n",
    "\n",
    "    pathMat   <- gmtListToSparseMat(gmt_list)\n",
    "    matched   <- getMatchedPathwayMat(pathMat, gtex_genes)\n",
    "    if (is.null(matched) || ncol(matched) == 0) {\n",
    "      warning(\"No matched pathways for \", prior_name, \"; skipping.\")\n",
    "      next\n",
    "    }\n",
    "    chatObj   <- getChat(matched)\n",
    "\n",
    "    fullRes <- PLIERfull(\n",
    "      Y                 = gtex_fbm_filt,\n",
    "      priorMat          = as.matrix(matched),\n",
    "      svdres            = gtex_svdRes,\n",
    "      plier.base.result = gtex_baseRes,\n",
    "      Chat              = chatObj,\n",
    "      k                 = PLIER_K_gtex,\n",
    "      doCrossval        = TRUE,\n",
    "      trace             = TRUE,\n",
    "      max.U.updates     = config$GTEx$MAX_U_UPDATES\n",
    "    )\n",
    "\n",
    "    if (!is.null(fullRes$B)) colnames(fullRes$B) <- samples\n",
    "    if (!is.null(fullRes$Z)) colnames(fullRes$Z) <- paste0(\"LV\", seq_len(ncol(fullRes$Z)))\n",
    "    if (!is.null(fullRes$summary)) {\n",
    "      fullRes$summary <- fullRes$summary |>\n",
    "        dplyr::rename(LV = LV_index) |>\n",
    "        dplyr::mutate(LV = paste0(\"LV\", LV))\n",
    "    }\n",
    "\n",
    "    out_file <- file.path(output_data_dir, sprintf(\"gtex_%s_PLIER2.rds\", prior_name))\n",
    "    saveRDS(fullRes, file = out_file)\n",
    "    results[[prior_name]] <- fullRes\n",
    "\n",
    "    message(\"Saved: \", out_file)\n",
    "  }, silent = FALSE)\n",
    "}\n",
    "\n",
    "invisible(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccad0fa0",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Auto-detected name: KEGG_2021_Human\n",
      "\n",
      "Using cached file for KEGG_2021_Human\n",
      "\n",
      "Auto-detected name: GWAS_Catalog_2025\n",
      "\n",
      "Using cached file for GWAS_Catalog_2025\n",
      "\n",
      "Auto-detected name: GO_Biological_Process_2025\n",
      "\n",
      "Using cached file for GO_Biological_Process_2025\n",
      "\n",
      "Auto-detected name: GTEx_Tissues_V8_2023\n",
      "\n",
      "Using cached file for GTEx_Tissues_V8_2023\n",
      "\n",
      "Auto-detected name: Chromosome_Location\n",
      "\n",
      "Using cached file for Chromosome_Location\n",
      "\n",
      "Auto-detected name: Human_Gene_Atlas\n",
      "\n",
      "Using cached file for Human_Gene_Atlas\n",
      "\n",
      "Auto-detected name: LINCS_L1000_CRISPR_KO_Consensus_Sigs\n",
      "\n",
      "Using cached file for LINCS_L1000_CRISPR_KO_Consensus_Sigs\n",
      "\n",
      "Auto-detected name: Metabolomics_Workbench_Metabolites_2022\n",
      "\n",
      "Using cached file for Metabolomics_Workbench_Metabolites_2022\n",
      "\n",
      "Auto-detected name: OMIM_Disease\n",
      "\n",
      "Using cached file for OMIM_Disease\n",
      "\n",
      "Auto-detected name: Proteomics_Drug_Atlas_2023\n",
      "\n",
      "Using cached file for Proteomics_Drug_Atlas_2023\n",
      "\n",
      "Auto-detected name: TF_Perturbations_Followed_by_Expression\n",
      "\n",
      "Using cached file for TF_Perturbations_Followed_by_Expression\n",
      "\n",
      "Auto-detected name: UK_Biobank_GWAS_v1\n",
      "\n",
      "Using cached file for UK_Biobank_GWAS_v1\n",
      "\n",
      "there are  16450  genes in the intersection between data and prior\n",
      "\n",
      "Removing 3828 pathways\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 3.3 GiB”\n",
      "Inverting...\n",
      "\n",
      "done\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 3.3 GiB”\n",
      "**PLIER v2 **\n",
      "\n",
      "using provided PLIERbase result\n",
      "\n",
      "L1=67.1332478272621; L2=201.399743481786\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress 81 / 350 | Bdiff=0.000191 Number of annotated columns is 163, Number of annotated columns is 163, Number of annotated columns is 164, Number of annotated columns is 165, Number of annotated columns is 167, Number of annotated columns is 162, Number of annotated columns is 162, Number of annotated columns is 164, Number of annotated columns is 165, Number of annotated columns is 165, Number of annotated columns is 163, Number of annotated columns is 164, Number of annotated columns is 164, Number of annotated columns is 165, Number of annotated columns is 167, Number of annotated columns is 164, Number of annotated columns is 165, Number of annotated columns is 168, Number of annotated columns is 168, Number of annotated columns is 170, Number of annotated columns is 167, Number of annotated columns is 167, Number of annotated columns is 168, Number of annotated columns is 170, Number of annotated columns is 172, Number of annotated columns is 169, Number of annotated columns is 170, Number of annotated columns is 171, Number of annotated columns is 172, Number of annotated columns is 173, Number of annotated columns is 170, Number of annotated columns is 170, Number of annotated columns is 170, Number of annotated columns is 170, Number of annotated columns is 170, Number of annotated columns is 168, Number of annotated columns is 169, Number of annotated columns is 169, Number of annotated columns is 170, Number of annotated columns is 170"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converged at  iteration 81 Bdiff is not decreasing\n",
      "\n",
      "Updating Z for CV\n",
      "\n",
      "crossValidation\n",
      "\n",
      "There are 105  LVs with AUC>0.70\n",
      "\n",
      "There are 60  LVs with AUC>0.90\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# priors\n",
    "gtex_gmtList <- list(\n",
    "  KEGG = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=KEGG_2021_Human\"),\n",
    "  GWAS_Catalog = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GWAS_Catalog_2025\"),\n",
    "  BP = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GO_Biological_Process_2025\"),\n",
    "  GTEx_Tissues = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=GTEx_Tissues_V8_2023\"),\n",
    "  Chromosome_Location = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Chromosome_Location\"),\n",
    "  Human_Gene_Atlas = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Human_Gene_Atlas\"),\n",
    "  LINCS_L1000_CRISPR_KO = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=LINCS_L1000_CRISPR_KO_Consensus_Sigs\"),\n",
    "  Metabolomics_Workbench = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Metabolomics_Workbench_Metabolites_2022\"),\n",
    "  OMIM_Disease = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=OMIM_Disease\"),\n",
    "  Proteomics_Drug_Atlas = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=Proteomics_Drug_Atlas_2023\"),\n",
    "  TF_Perturbations = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=TF_Perturbations_Followed_by_Expression\"),\n",
    "  UK_Biobank_GWAS = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=UK_Biobank_GWAS_v1\"),\n",
    "  CellMarker = getGMT(\"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&libraryName=CellMarker_2024\")\n",
    "\n",
    ")\n",
    "\n",
    "# prefix each gene‐set name with its library to guarantee uniqueness\n",
    "for(lib in names(gtex_gmtList)) {\n",
    "  names(gtex_gmtList[[lib]]) <- paste0(lib, \"_\", names(gtex_gmtList[[lib]]))\n",
    "}\n",
    "\n",
    "gtex_pathMat <- gmtListToSparseMat(gtex_gmtList)\n",
    "gtex_matched <- getMatchedPathwayMat(gtex_pathMat, gtex_genes)\n",
    "gtex_chatObj <- getChat(gtex_matched)\n",
    "\n",
    "# PLIERfull\n",
    "gtex_fullRes <- PLIERfull(\n",
    "  Y                 = gtex_fbm_filt,\n",
    "  priorMat          = as.matrix(gtex_matched),\n",
    "  svdres            = gtex_svdRes,\n",
    "  plier.base.result = gtex_baseRes,\n",
    "  Chat              = gtex_chatObj,\n",
    "  k                 = PLIER_K_gtex,\n",
    "  doCrossval        = TRUE,\n",
    "  trace             = TRUE,\n",
    "  max.U.updates= config$GTEx$MAX_U_UPDATES\n",
    ")\n",
    "\n",
    "# Fix colnames and rownames\n",
    "colnames(gtex_fullRes$B) <- samples\n",
    "colnames(gtex_fullRes$Z) <- paste0('LV', seq_len(ncol(gtex_fullRes$Z)))\n",
    "gtex_fullRes$summary <- gtex_fullRes$summary %>%\n",
    "    dplyr::rename(LV = LV_index)  %>% \n",
    "    dplyr::mutate(LV = paste0('LV', LV))\n",
    "\n",
    "# save\n",
    "saveRDS(gtex_fullRes, file = file.path(output_data_dir, \"gtex_all_PLIER2.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2478e517",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gtex_GO_BP_PLIER2 <- readRDS(here(\"output/gtex/gtex_GO_Biological_Process_2025_PLIER2.rds\"))\n",
    "gtex_KEGG_PLIER2 <- readRDS(here(\"output/gtex/gtex_KEGG_2021_Human_PLIER2.rds\"))\n",
    "gtex_GTEx_Tissues_PLIER2 <- readRDS(here(\"output/gtex/gtex_GTEx_Tissues_V8_2023_PLIER2.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "aa1af70d",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "models <- list(\n",
    "    gtex_GO_BP_PLIER2 = gtex_GO_BP_PLIER2,\n",
    "    gtex_KEGG_PLIER2 = gtex_KEGG_PLIER2,\n",
    "    gtex_GTEx_Tissues_PLIER2 = gtex_GTEx_Tissues_PLIER2\n",
    ")\n",
    "\n",
    "for (model_name in names(models)) {\n",
    "    model <- models[[model_name]]\n",
    "    model_dir <- file.path(output_data_dir, model_name)\n",
    "    dir.create(model_dir, showWarnings = FALSE, recursive = TRUE)\n",
    "    \n",
    "    B <- as.data.frame(model$B)\n",
    "    write.csv(B, file.path(model_dir, \"gtex_B.csv\"), row.names = TRUE)\n",
    "    \n",
    "    Z <- as.data.frame(model$Z)\n",
    "    write.csv(Z, file.path(model_dir, \"gtex_Z.csv\"), row.names = TRUE)\n",
    "    \n",
    "    summary <- model$summary\n",
    "    write.csv(summary, file.path(model_dir, \"gtex_summary.csv\"), row.names = FALSE)\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
